test/inductor/test_codecache.py

```diff
@@ -494,13 +494,41 @@ def fn2(q, k, v):
     @config.patch({"fx_graph_cache": True})
     @config.patch({"fx_graph_remote_cache": False})
     @parametrize("bundle_triton", (False, True))
-    @parametrize("grad", (False, True))
-    def test_triton_higher_order_op_bypass(self, bundle_triton, grad):
+    def test_higher_order_op_bypass(self, bundle_triton):
         """
-        Verify that we bypass the cache when we have a triton higher order ops
+        Verify that we bypass the cache when we have a higher order ops
         and that bundler start/end works with a cache bypass.
         """
 
+        def fn(x):
+            def true_fn(x: torch.Tensor):
+                return x.cos()
+
+            def false_fn(x: torch.Tensor):
+                return x.sin()
+
+            return torch.cond(x.shape[0], true_fn, false_fn, (x,))
+
+        with config.patch(bundle_triton_into_fx_graph_cache=bundle_triton):
+            compiled_fn = torch.compile(fn, dynamic=True, fullgraph=True)
+
+            x = torch.randn(4, 4, device=GPU_TYPE)
+            result = compiled_fn(x)
+
+            self.assertEqual(counters["inductor"]["fxgraph_cache_miss"], 0)
+            self.assertEqual(counters["inductor"]["fxgraph_cache_hit"], 0)
+            self.assertGreater(counters["inductor"]["fxgraph_cache_bypass"], 0)
+
+    @requires_gpu()
+    @requires_triton()
+    @config.patch({"fx_graph_cache": True})
+    @config.patch({"fx_graph_remote_cache": False})
+    @parametrize("bundle_triton", (False, True))
+    def test_triton_higher_order_op(self, bundle_triton):
```

Comment:
For testing:
- can we test that calling a function with a different triton kernel results in a cache miss?